File: src/main/java/de/bushnaq/abdalla/kassandra/report/dao/theme/BurndownTheme.java
/*
 *
 * Copyright (C) 2025-2026 Abdalla Bushnaq
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

package de.bushnaq.abdalla.kassandra.report.dao.theme;

import java.awt.*;

public class BurndownTheme {
    public static final int     MAX_AUTHOR_COLOR = 12/*(9-4) * 3*//*+ 3*//*+1*/;
    //    public              Color   bankHolidayColor;
    public              Color   burnDownBorderColor;
    public              Color[] burnDownColor;
    public              Color   delayEventColor;
    public              Color   inTimeColor;
    public              Color   optimaleGuideColor;
    public              Color   plannedGuideColor;
    public              Color   tickTextColor;
    public              Color   ticksColor;
    public              Color   watermarkColor;

    public BurndownTheme() {
        this.burnDownColor = new Color[MAX_AUTHOR_COLOR];
    }

    public Color getAuthorColor(int index) {
        if (index < 0) {
            index = -index;
        }
        int colorIndex = index % MAX_AUTHOR_COLOR;
        return burnDownColor[colorIndex];
    }
}

File: src/main/frontend/js/theme/burndown-theme.ts
// theme/burndown-theme.ts
// Mirrors Java: BurndownTheme (including burnDownColor[12] array)
//
// Copyright (C) 2025-2026 Abdalla Bushnaq – Apache License 2.0

export class BurndownTheme {
    readonly borderColor:        number | null;
    /** Array of 12 author colors. Mirrors Java BurndownTheme.burnDownColor[]. */
    readonly burnDownColor:      (number | null)[];
    readonly delayEventColor:    number | null;
    readonly inTimeColor:        number | null;
    readonly optimaleGuideColor: number | null;
    readonly plannedGuideColor:  number | null;
    readonly tickTextColor:      number | null;
    readonly ticksColor:         number | null;
    readonly watermarkColor:     number | null;

    constructor(data: Record<string, unknown> = {}) {
        this.borderColor        = (data['borderColor']        as number | null) ?? null;
        this.burnDownColor      = Array.isArray(data['burnDownColor'])
            ? (data['burnDownColor'] as (number | null)[]).slice()
            : new Array<number | null>(12).fill(null);
        this.delayEventColor    = (data['delayEventColor']    as number | null) ?? null;
        this.inTimeColor        = (data['inTimeColor']        as number | null) ?? null;
        this.optimaleGuideColor = (data['optimaleGuideColor'] as number | null) ?? null;
        this.plannedGuideColor  = (data['plannedGuideColor']  as number | null) ?? null;
        this.tickTextColor      = (data['tickTextColor']      as number | null) ?? null;
        this.ticksColor         = (data['ticksColor']         as number | null) ?? null;
        this.watermarkColor     = (data['watermarkColor']     as number | null) ?? null;
    }

    /** Mirrors Java: BurndownTheme.getAuthorColor(int index) */
    getAuthorColor(index: number): number | null {
        const i = index < 0 ? -index : index;
        return this.burnDownColor[i % 12] ?? null;
    }
}

